.github/workflows/build.yml

```diff
@@ -56,21 +58,24 @@ jobs:
   # Build step.
   build:
     name: "Gradle builder"
-    runs-on: ubuntu-latest
-
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        java-version: [ 16 ]
+        distro: [ 'zulu', 'temurin' ]
     # Will run only if analyze_sources determined it is needed.
     needs: analyze_sources
     if: needs.analyze_sources.outputs.run_build == 'true'
 
     steps:
     - name: "Checkout sources"
       uses: actions/checkout@v2
-
-    - name: "Set up JDK"
+    - name: Set up JDK ${{ matrix.java-version }} ${{ matrix.distro }}
```

Comment:
Please do not use "-" in variable names even if supported. Please replace with underscore instead. Also it's JDK version rather than Java version as former var name indicated.

Reply:
will do. I will make it the former with all caps.

Reply:
Can you please also D.R.Y. and make both JDK version and distro name variables first (can be env) and then use these wars wherever needed in `*.yml`? ALL_CAPS is fine in case of global or env vars.